lib/evm/operation/duplication.ex

```diff
@@ -5,11 +5,11 @@ defmodule EVM.Operation.Duplication do
 
   ## Examples
 
-      iex> EVM.Operation.Duplication.dup_n(2, [5], %{})
-      [5, 5]
+      iex> EVM.Operation.Duplication.dup([1, 2, 3], %{})
+      [3, 2, 1, 1]
   """
-  @spec dup_n(integer(), Operation.stack_args, Operation.vm_map) :: Operation.op_result
-  def dup_n(n, [s0], _) do
-    for _ <- 1..n, do: s0
+  @spec dup(Operation.stack_args, Operation.vm_map) :: Operation.op_result
+  def dup(list = [ head | _ ], _) do
+    :lists.reverse(list) ++ [head]
```

Comment:
I don't quite understand, -- why do we end up reversing the args to get the proper output?

Reply:
Yeah, it's a little funky. I think it's because we're popping off the top of the stack and then pushing the same stack back into place, like if you drew 5 cards and then put them back in the same order they were in. I tried doing the reversing in `Stack.push` but that got even more confusing :/